rust/definitions/src/navigation.rs

```diff
@@ -0,0 +1,661 @@
+use crate::{
+    crypto::Encryption, history::Event, keyring::NetworkSpecsKey, network_specs::NetworkSpecs,
+};
+
+pub use crate::network_specs::NetworkSpecsToSend;
+
+#[derive(PartialEq, Clone)]
+pub struct SeedNameWithIdenticon {
+    pub seed_name: String,
+    pub identicon: Vec<u8>,
+}
+
+/// Enum containing card sets for three different outcomes:
+/// signing (Sign), accepting (Stub) and reading, for example, in case of an error (Read)
+#[derive(PartialEq, Debug, Clone)]
+pub enum TransactionAction {
+    Derivations {
+        content: TransactionCardSet,
+        network_info: NetworkSpecs,
+        checksum: u32,
+        network_specs_key: NetworkSpecsKey,
+    },
+    Sign {
+        content: TransactionCardSet,
+        checksum: u32,
+        has_pwd: bool,
+        author_info: Address,
+        network_info: NetworkSpecs,
+    },
+    Stub {
+        s: TransactionCardSet,
+        u: u32,
+        stub: StubNav,
+    },
+    Read {
+        r: TransactionCardSet,
+    },
+}
+
+/// Enum describing Stub content.
+/// Is used for proper navigation. Variants:
+/// AddSpecs (with associated NetworkSpecsKey), LoadMeta (with associated
+/// NetworkSpecsKey for the first by order network using those metadata),
+/// and LoadTypes
+#[derive(PartialEq, Debug, Clone)]
+pub enum StubNav {
+    AddSpecs { n: NetworkSpecsKey },
+    LoadMeta { l: NetworkSpecsKey },
+    LoadTypes,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum FooterButton {
+    Log,
+    Scan,
+    Keys,
+    Settings,
+    Back,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum RightButton {
+    LogRight,
+    NewSeed,
+    Backup,
+    MultiSelect,
+    NDMenu,
+    TypesInfo,
+    KeyMenu,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum ScreenNameType {
+    H1,
+    H4,
+}
+
+#[derive(PartialEq, Debug, Clone)]
+pub enum ShieldAlert {
```

Comment:
Explanation on this alert.

Rust side knows only whether the shield alarm was triggered or reset. Thus, values here should be only `past` or `none` (or `safe` or something like that). Source of truth for this is database, alert is set to `past` when actual state moves to `active` and set to `none` when acknowledge button is pressed in shield modal from state `past`

The frontend controls whether `active` state is triggered in shortest possible path. `active` state triggers on network detection and clears on network loss. Only triggering to 'active' sends signal to rust side.

Thus, logic in UI is something like:

```
if (alert) {
  ShowDisconnectNetwork()
} else {
  if (past) {
    ShowAlertWithAcknowledgeButton()
  } else {
    ShowAllIsSafe()
  }
}

Reply:
what is `if(alert)` what exactly are you proposing to match on then?

Reply:
how exactly in types are you proposing to move it into `active` state when you are requesting to remove the `active` variant from the type?

Reply:
> what is `if(alert)` what exactly are you proposing to match on then?

corrected



> how exactly in types are you proposing to move it into `active` state when you are requesting to remove the `active` variant from the type?

It will only exist in UI, as it does in released version. It pretty much is just a check for network availability that is completely on native framework side.